spec/compiler/formatter/formatter_spec.cr

```diff
@@ -1610,4 +1610,8 @@ describe Crystal::Formatter do
       # second comment
     }
     CODE
+
+  # format expressions in `# =>` comments
+  assert_format "1 # => ?invalid code"
+  assert_format "1 # => #=>1+2", "1 # => # => 1 + 2"
```

Comment:
These specs cover known edge cases but it's as important to test a few basic cases first.

Reply:
What basic cases would you suggest? The two cases that need to be covered are invalid and valid code. Invalid code is covered by the first case, for valid code I could add another basic case like `1 # => 1 + 2` but that is also already covered by the recursion in case two.

Reply:
What. happens if the comment expression isa method definition with, say, 1 + 2 as the body?

Reply:
That is indeed a problem I did not consider. As it is now, the formatter will put the body and `end` in separate lines which are not commented out:
```crystal
1 # => def add12() 1 + 2 end
```
turns into
```crystal
1 # => def add12
  1 + 2
end
```
It would be a good idea, anyway, to only apply formatting to certain kinds of expressions, or better still, to only apply formatting if the formatted string is equal to the result of `.inspect`ing the expression before the comment (see discussion in #7272). If that was implemented, the above problem wouldn't even come up. So before fixing this, it might be a good idea to overthink the whole approach here...

Reply:
I think it's simpler: try to format it. If you find newlines, then don't use the result.

Reply:
Of course that means that we can't always require what comes after `# => ` to be properly formatted. /cc @straight-shoota

Reply:
If the formatter inserts a line break, that looks like an indicator that there is something wrong. I'm convinced it should fail in that case. I can't think of any example where the proper result of an expression described with ` =>` would be split in multiple lines.

Reply:
I don't think the code formatter should ever fail. It would be pretty annoying...

Reply:
Specially considering that it's a tool that's combined with the editor. I would find it pretty annoying if the editor starts spitting errors at me because of the way I organize my code.

Reply:
Yeah maybe the formatter is not the right floor this. But it would be great to have some formal validation for `# =>` comments and similar.
Maybe that's something for ameba?

Reply:
If anything, a tool like doctest would be great.

I think having the formatter try to format the value if possible is good enough. Then doctest would actually run it and verify that the results are what you'd expect them to be.

Reply:
Incidentally, is there a reason why `# def add12 : Int32 1 + 2 end` and `# def add12() 1 + 2 end` are valid syntax while `# def add12 1 + 2 end` raises the syntax error "unexpected token: 1"?

Reply:
Just a lack of thought from us on those details. We should probably copy Ruby or decide what to do, but it's very low priority.